### PR TITLE
raftstore: fix load base split cannot works in pure follower/stale read scenario

### PR DESCRIPTION
**Pull request by nolouch**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/15539

What's Changed:

I try to redirect the split command in tikv layer but found it not easy to do because the raftcommand is not designed to send between raft nodes. But using `RaftMessage` should add a new raft message and make it more complex.
On the other hand, the proposal can be a propose on the follower node, and the raft layer will redirect:

https://github.com/tikv/raft-rs/blob/65a00620d80a89603c3358fd4b90265a692ae767/src/raft.rs#L2302-L2315

```commit-message
raftstore: fix load base split cannot works in pure follower/stale read scenario
- allow split command proposal on non leader peer
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/fsm/peer.rs [hunk 1]
```diff
@@ -5321,10 +5321,14 @@ where
...
         let flags = WriteBatchFlags::from_bits_check(msg.get_header().get_flags());
         let allow_stale_read = read_only && flags.contains(WriteBatchFlags::STALE_READ);
+        let split_region = msg.has_admin_request()
+            && msg.get_admin_request().get_cmd_type() == AdminCmdType::BatchSplit;
         if !self.fsm.peer.is_leader()
             && !is_read_index_request
             && !allow_replica_read
             && !allow_stale_read
+            // allow proposal split command at non leader, raft layer will forward it to leader.
+            && !split_region
```
Comment (overvenus) [inline components/raftstore/src/store/fsm/peer.rs:5331]: Could you add a metric for redirected raft proposal?

**Review by overvenus — APPROVED** [review]
Rest LGTM

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**